students/PadmajaPeri/lesson04/kata_assignment.py

```diff
@@ -0,0 +1,57 @@
+#/usr/bin/env python3
+
+"""
+Program to generate new text given an input text. It uses a dictionary to
+build  trigrams of current text.
+__author__ == 'Padmaja Peri'
+"""
+
+import random
+
+
+def build_trigram_dict(fd):
+    """ Method to build dictionary of trigrams from a text file """
+    words_dict = {}
+    text = fd.read().split()
+    idx = 0
+    # Stop looping when we have only two words left
+    while idx < len(text) - 2:
+        first_word = text[idx]
+        second_word = text[idx + 1]
+        third_word = text[idx + 2]
+        dict_key = first_word + ' ' + second_word
```

Comment:
you could probably have just made this `dict_key = text[idx] + ' ' + text[idx + 1]` instead of having the extra variables